pkg/operator/controller/ingress/status.go

```diff
@@ -979,7 +979,14 @@ func computeDNSStatus(ic *operatorv1.IngressController, wildcardRecord *iov1.DNS
 		}
 	}
 	var conditions []operatorv1.OperatorCondition
-	if ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS {
+	if !manageDNSForDomain(ic.Status.Domain, status, dnsConfig) && ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS {
+		conditions = append(conditions, operatorv1.OperatorCondition{
```

Comment:
Should we just return the single condition `return []operatorv1.OperatorCondition` here?  Do we really want to go on and check the `wildcardRecord` and append more conditions after this?

Reply:
The two sections are for 2 different condition types (`DNSManagedIngressConditionType` and `DNSReadyIngressConditionType`), so even if domain doesn't match and ingresscontroller status has `dnsManagementPolicy` as `Unmanaged` we still need to update `DNSReady` condition based on wildcard record.